File: schemaorggo/datafeeditem.go
package schemaorggo

import "encoding/json"

// DataFeedItem see : https://schema.org/DataFeedItem
type DataFeedItem struct {
	Intangible

	typeContext

	// DateCreated see : https://schema.org/dateCreated
	// The date on which the CreativeWork was created or the item was added to a DataFeed.
	// types : Date DateTime
	DateCreated []interface{} `json:"dateCreated,omitempty"`

	// DateDeleted see : https://schema.org/dateDeleted
	// The datetime the item was removed from the DataFeed.
	// types : DateTime
	DateDeleted []DateTime `json:"dateDeleted,omitempty"`

	// DateModified see : https://schema.org/dateModified
	// The date on which the CreativeWork was most recently modified or when the item&#39;s entry was modified within a DataFeed.
	// types : Date DateTime
	DateModified []interface{} `json:"dateModified,omitempty"`

	// Item see : https://schema.org/item
	// An entity represented by an entry in a list or data feed (e.g. an &#39;artist&#39; in a list of &#39;artists&#39;)’.
	// types : Thing
	Item []*Thing `json:"item,omitempty"`
}

func (v DataFeedItem) IntoMap(intop *map[string]interface{}) error {
	if intop == nil {
		return nil
	}

	v.Intangible.IntoMap(intop)

	into := *intop

	{
		var value interface{} = v.DateCreated
		if len(v.DateCreated) == 1 {
			value = v.DateCreated[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["dateCreated"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.DateDeleted
		if len(v.DateDeleted) == 1 {
			value = v.DateDeleted[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["dateDeleted"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.DateModified
		if len(v.DateModified) == 1 {
			value = v.DateModified[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["dateModified"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.Item
		if len(v.Item) == 1 {
			value = v.Item[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["item"] = json.RawMessage(b)
		}
	}

	*intop = into

	return nil
}

func (v DataFeedItem) AsMap() (map[string]interface{}, error) {
	data := map[string]interface{}{}
	err := v.IntoMap(&data)
	if err != nil {
		return nil, err
	}

	data["@context"] = "http://schema.org"
	data["@type"] = "DataFeedItem"

	return data, nil
}

func (v DataFeedItem) MarshalJSON() ([]byte, error) {
	data, err := v.AsMap()
	if err != nil {
		return nil, err
	}

	return json.Marshal(data)
}

File: schemaorgphp/classes/data_feed_item.php
<?php

class DataFeedItem extends Intangible implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'DataFeedItem';
	
	/**
	 * The date on which the CreativeWork was created or the item was added to a DataFeed.
	 * see : https://schema.org/dateCreated
	 * @var string|string[]|string|string[]
	 */
	public var $date_created;
	
	/**
	 * The datetime the item was removed from the DataFeed.
	 * see : https://schema.org/dateDeleted
	 * @var string|string[]
	 */
	public var $date_deleted;
	
	/**
	 * The date on which the CreativeWork was most recently modified or when the item&#39;s entry was modified within a DataFeed.
	 * see : https://schema.org/dateModified
	 * @var string|string[]|string|string[]
	 */
	public var $date_modified;
	
	/**
	 * An entity represented by an entry in a list or data feed (e.g. an &#39;artist&#39; in a list of &#39;artists&#39;)’.
	 * see : https://schema.org/item
	 * @var \Thing|\Thing[]
	 */
	public var $item;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'DataFeedItem'
		);
		
		$serialized = so_json_serialize( $this->date_created );
		if ( ! empty( $serialized ) ) {
			$out['dateCreated'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->date_deleted );
		if ( ! empty( $serialized ) ) {
			$out['dateDeleted'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->date_modified );
		if ( ! empty( $serialized ) ) {
			$out['dateModified'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->item );
		if ( ! empty( $serialized ) ) {
			$out['item'] = $serialized;
		}
		
		return $out;
	}
}
